Handle installed Scarb version (#243)
dojoengine/book: docs/pages/getting-started/quick-start.md

For full `dojoup` reference and debugging see [Dojoup](/toolchain/dojoup.md).

### Install scarb (optional)

For the language server to work properly, your local Scarb version should not be lower than Dojo's cairo version. To accomplish this, install it with the following command:

```sh

> # todo: input command here

Reviewer: Instead of the script, we may give the user the commands to run for him to see both versions: the dojo one, and the local one.
WDYT?

Author: Sure. Gonna try it.

Author: I show the user the commands to see scarb local version and installation of scarb on commit [be0b3ed](https://github.com/dojoengine/book/pull/243/commits/be0b3ed888d8899fe0d414ab118ddc90c5fa6bc2)

I'm uncertain which command to use to display the version of Cairo used by Dojo @glihm

Reviewer: > I show the user the commands to see scarb local version and installation of scarb on commit [be0b3ed](https://github.com/dojoengine/book/pull/243/commits/be0b3ed888d8899fe0d414ab118ddc90c5fa6bc2)
> 
> I'm uncertain which command to use to display the version of Cairo used by Dojo @glihm I am testing it with dojo lastest alpha 0.7 release

That's actually a good point. We should add to `sozo --version` something to output the actual cairo version that is supported by Dojo. This should be the easiest way to display such information and to ensure the user can easily install it.

Author: I can create a PR for that

For full `dojoup` reference and debugging see [Dojoup](/toolchain/dojoup.md).

### Install Scarb (optional)

> To ensure the proper functioning of the language server, your local `Scarb` version must be equal to or higher than the Dojo's Cairo version. To accomplish this, first check your Dojo's Cairo version. To achieve this, run the following command:

Reviewer: ```suggestion
To ensure the proper functioning of the language server, your local `Scarb` version must be equal to or higher than the Dojo's Cairo version. To accomplish this, first check your Dojo's Cairo version running the following command:
```

```bash
sozo --version
```

> You might expect an output similar to this:

Reviewer: ```suggestion
You should have an output similar to this:
```

```console
sozo 0.6.0-alpha.9
scarb: 2.5.4
cairo: 2.5.4
sierra: 1.4.0
```

After that, check your local `Scarb` version:

```sh

> scarb --version

Reviewer: Also show the output to guide the user as you did for sozo.

```

> If your local `Scarb` version is lower than the Dojo's Cairo version, you have to install `Scarb`.

Reviewer: ```suggestion
If your local `Scarb` version is lower than the Dojo's Cairo version, you have to install a newer version `Scarb`.
```

To accomplish it, please refer to the [installation instructions](https://docs.swmansion.com/scarb/download).
We strongly recommend that you install
Scarb via [asdf](https://docs.swmansion.com/scarb/download.html#install-via-asdf), a CLI tool that can manage
multiple language runtime versions on a per-project basis.
This will ensure that the version of Scarb you use to work on a project always matches the one defined in the
project settings, avoiding problems related to version mismatches.

Please refer to the [asdf documentation](https://asdf-vm.com/guide/getting-started.html) to install all
prerequisites.

Once you have `asdf` installed locally, you can download Scarb plugin with the following command:

```bash
asdf plugin add scarb
```

> This will allow you to download specific versions. You can choose the same version as the Dojo's Cairo version, for example, 2.5.4, with the following command:

Reviewer: ```suggestion
This will allow you to download specific versions. You can choose the same version as the Dojo's Cairo version, for example, `2.5.4`, with the following command:
```

sierra: 1.4.0
```

If your local `Scarb` version is lower than the Dojo's Cairo version, you have to install a newer version `Scarb`.

To accomplish it, please refer to the [installation instructions](https://docs.swmansion.com/scarb/download).
We strongly recommend that you install
Scarb via [asdf](https://docs.swmansion.com/scarb/download.html#install-via-asdf), a CLI tool that can manage
multiple language runtime versions on a per-project basis.
This will ensure that the version of Scarb you use to work on a project always matches the one defined in the
project settings, avoiding problems related to version mismatches.

Please refer to the [asdf documentation](https://asdf-vm.com/guide/getting-started.html) to install all
prerequisites.

Once you have `asdf` installed locally, you can download Scarb plugin with the following command:

```bash
asdf plugin add scarb
```

This will allow you to download specific versions. You can choose the same version as the Dojo's Cairo version, for example, `2.5.4`, with the following command:

```bash
asdf install scarb 2.5.4
```

and set a global version:

```bash
asdf global scarb 2.5.4
```

> Otherwise, you can simply run the following command in your terminal, and follow the onscreen instructions. This

Reviewer: I would just add an example of installing a specific version with curl. And we're good to merge. :pray:
